### DISCUSSION TITLE
actix_service::AppRouting should be public

### DISCUSSION THREAD

**Discussion by Necoo33** [actix/actix-web#3491] [Q&A] [answered]
Hello, i'm trying to implement an authentication flow, in that moment i'm coding a cookie refresh middleware in v4.9 and with using `actix_web::app::App::wrap` function i encountered an issue. Because of that `actix_web::actix_service::AppRouting` is not public i cannot put it inside a wrapper function, which that makes my code messier. my wrap function remained like this:

```rust

// other services
            .wrap_fn(move |req, srv| {
                let auth_cookie_name = match env::var("AUTH_COOKIE_NAME") {
                    Ok(name) => name,
                    Err(_) => "auth_cookie".to_string()
                };

                let cookie = req.cookie(&auth_cookie_name);

                let fut = srv.call(req);
            
                async move {
                    let mut res = fut.await?;
            
                    res = utils::refresh_cookies(cookie, auth_cookie_name, res).await.unwrap();
            
                    Ok(res)
                }
            })

````

so i think in next version you should make public the second argument of the inner closure of .wrap() method.

**Comment by robjtede** [comment, marked as answer]
see:
- https://github.com/actix/actix-web/issues/3033#issuecomment-1544127642
- https://github.com/actix/actix-web/wiki/FAQ#how-can-i-return-app-from-a-function--why-is-appentry-private
